lightning/src/ln/channelmanager.rs

```diff
@@ -2559,14 +2552,19 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 	/// Value parameters are provided via the last hop in route, see documentation for RouteHop
 	/// fields for more info.
 	///
-	/// Note that if the payment_hash already exists elsewhere (eg you're sending a duplicative
-	/// payment), we don't do anything to stop you! We always try to ensure that if the provided
-	/// next hop knows the preimage to payment_hash they can claim an additional amount as
-	/// specified in the last hop in the route! Thus, you should probably do your own
-	/// payment_preimage tracking (which you should already be doing as they represent "proof of
-	/// payment") and prevent double-sends yourself.
+	/// If a pending payment is currently in-flight with the same [`PaymentId`] provided, this
+	/// method will error with an [`APIError::RouteError`]. Note, however, that once a payment
+	/// is no longer pending (either via [`ChannelManager::abandon_payment`], or handling of an
+	/// [`Event::PaymentSent`]) a second payment with the same [`PaymentId`] can be sent.
+	///
+	/// Thus, in order to ensure duplicate payments are not sent, you should implement your own
+	/// tracking of payments, including state to indicate once a payment has completed. Because you
+	/// should also ensure that [`PaymentHash`]es are not re-used, for simplicity, you should
+	/// consider using the [`PaymentHash`] as the key for tracking payments. In that case, the
```

Comment:
can we get away with this recommendation "consider using the [`PaymentHash`] as the key for tracking payments."? 
Even though its a recommendation but seems like a restriction on client architecture.

Is it possible to use their contextId i.e paymentId for payment tracking ? 
and some random uuid kind of payment hash

Reply:
IIUC, 
It is possible to use contextId i.e paymentId as key for payment tracking and all data, and some random uuid kind of payment hash every time. 

The following line could be:
"for simplicity, you should consider using the [`PaymentHash`] /  [`PaymentId`] as the key for tracking payments."